adapters/telegram/src/polling.ts

```diff
@@ -29,13 +30,30 @@ export class HttpPolling extends Adapter.Client<TelegramBot> {
           return bot.offline()
         }
 
-        const updates = await bot.internal.getUpdates({
-          offset: this.offset + 1,
-          timeout: bot.config.pollingTimeout,
-        })
-        for (const e of updates) {
-          this.offset = Math.max(this.offset, e.update_id)
-          handleUpdate(e, bot)
+        try {
+          const updates = await bot.internal.getUpdates({
+            offset: this.offset + 1,
+            timeout: bot.config.pollingTimeout,
+          })
+          if (bot.status === 'reconnect') bot.status = 'online'
+          for (const e of updates) {
+            this.offset = Math.max(this.offset, e.update_id)
+            handleUpdate(e, bot)
+          }
+        } catch (e) {
+          bot.status = 'reconnect'
+          // Telegram error
+          if (axios.isAxiosError(e) && e.response && e.response.data) {
+            const errorBody = e.response.data
+            logger.error('failed to get updates. (%c) reason: %s %c', errorBody.error_code, errorBody.description, 'telegram: ' + bot.selfId)
```

Comment:
Use `warn` instead of `error`. Error is something more critical.

Reply:
I prefer `failed to get updates: (%s) %s` (without selfId).

1. This is not an occasion that needs to be highlighted with color
2. There are usually not multiple bots in development, while there are usually other ways to check the online status in production